src/middlewared/middlewared/api/base/server/ws_handler/rpc.py

```diff
@@ -264,11 +265,8 @@ async def validate_message(message: Any) -> None:
         except KeyError:
             raise ValueError("Missing 'jsonrpc' member")
 
-        try:
-            if not isinstance(message["id"], None | int | str):
-                raise ValueError("'id' member must be of type null, string or number")
-        except KeyError:
-            raise ValueError("Missing 'id' member")
+        if not isinstance(message.get("id"), None | int | str):
```

Comment:
This is a hot-path and it's quicker to access the member and catch the crash then it is to do `.get(`
```suggestion
          try:
              id_ = message["id"]
              if not isinstance(id_, None | int | str):
                  raise ValueError(...
          except KeyError:
              id_ = undefined
```

Reply:
How did you get this performance comparison? In my tests it's equal:
```
> timeit.timeit(stmt='isinstance(data.get("id"), None | int | str)', setup='data = {}')
0.10755996299985782
> timeit.timeit(stmt='try:\n\tisinstance(data["id"], None | int | str)\nexcept KeyError:\n\tpass', setup='data = {}')
0.1172254680000151
```

Reply:
I did both. Tested it the same way but I also generated flame graphs while CI was running against a system. Also, you're calling `message.get` twice (potentially 3 times) in your PR.

Would it be easier to add the `id` key to the message if it isn't provided similar to how I handle `params`?

Reply:
So `timeit` says those are the same, but looking at a flame graph one could conclude that one is faster than other?